torch/lowrank.py

```diff
@@ -0,0 +1,325 @@
+"""Implement various linear algebra algorithms for low rank matrices.
+"""
+
+import torch
+
+
+def is_sparse(A):
+    """Check if tensor A is a sparse tensor"""
+    if isinstance(A, torch.Tensor):
+        return A.layout == torch.sparse_coo
+    import scipy.sparse
+    return isinstance(A, scipy.sparse.base.spmatrix)
+
+
+def uniform(low=0.0, high=1.0, size=None, dtype=None, device=None):
+    """Returns a tensor filled with random numbers from a uniform
+    distribution on the interval :math:`[low, high)`
+
+    Arguments::
+
+      size (int...): a sequence of integers defining the shape of the
+        output tensor.  Can be a variable number of arguments or a
+        collection like a list or tuple.
+
+      dtype (:class:`torch.dtype`, optional): the desired data type of
+        returned tensor.  Default: if ``None``, uses a global default
+        (see :func:`torch.set_default_tensor_type`).
+
+      device (:class:`torch.device`, optional): the desired device of
+        returned tensor.  Default: if ``None``, uses the current
+        device for the default tensor type (see
+        :func:`torch.set_default_tensor_type`). :attr:`device` will be
+        the CPU for CPU tensor types and the current CUDA device for
+        CUDA tensor types.
+
+    """
+    attrs = dict(dtype=dtype, device=device)
+    if size is None:
+        r = low + (high - low) * torch.rand(1, **attrs)[0]
+    else:
+        r = low + (high - low) * torch.rand(*size, **attrs)
+    if dtype in [torch.complex32, torch.complex64, torch.complex128]:
+        if size is None:
+            i = low + (high - low) * torch.rand(1, **attrs)[0]
```

Comment:
nit: Is there a reason why `size` defaults to `None` instead of `1`?

Reply:
yes, if `size=1` then users might think that also `size=2`, for instance, is ok, but it is not: when the size is not given, then the return value should represent a scalar value.

Reply:
yes